smack-extensions/src/main/java/org/jivesoftware/smackx/softwareinfo/form/SoftwareInfoForm.java

```diff
@@ -0,0 +1,272 @@
+/**
+ *
+ * Copyright 2020 Aditya Borikar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.softwareinfo.form;
+
+import java.util.List;
+
+import org.jivesoftware.smack.packet.Element;
+import org.jivesoftware.smack.util.EqualsUtil;
+import org.jivesoftware.smack.util.HashCode;
+import org.jivesoftware.smackx.mediaelement.element.MediaElement;
+import org.jivesoftware.smackx.xdata.FormField;
+import org.jivesoftware.smackx.xdata.FormFieldChildElement;
+import org.jivesoftware.smackx.xdata.TextSingleFormField;
+import org.jivesoftware.smackx.xdata.form.FilledForm;
+import org.jivesoftware.smackx.xdata.form.Form;
+import org.jivesoftware.smackx.xdata.packet.DataForm;
+import org.jivesoftware.smackx.xdata.packet.DataForm.Item;
+import org.jivesoftware.smackx.xdata.packet.DataForm.Type;
+
+/**
+ * {@link Form} that contains the software information.
+ * <br>
+ * Instance of {@link SoftwareInfoForm} can be created using {@link Builder#build()} method.
+ * <br>
+ * To obtain an instance of {@link Builder}, use {@link SoftwareInfoForm#getBuilder()} method.
+ * <br>
+ * An example to illustrate is provided inside SoftwareInfoFormTest inside the test package.
+ */
+public final class SoftwareInfoForm extends FilledForm {
+
+    public static final String FORM_TYPE = "urn:xmpp:dataforms:softwareinfo";
+
+    private SoftwareInfoForm(DataForm dataForm) {
+        super(dataForm);
+    }
+
+    /**
+     * Returns name of the OS used by client.
+     * <br>
+     * @return os
+     */
+    public String getOS() {
+        return readFirstValue("os");
```

Comment:
The String "os" should be a static constant in the class and referenced at the two sites where it is used.